### PR TITLE
Add in various suggestions of people

### PR DESCRIPTION
**Pull request by Garanas**
Adds a hotkey to select all mass extractors that are upgrading.

Adds a hotkey to select all radars.

Adds a hotkey to filter your selection (of engineers) to one unit that represents the highest tech. All other engineers assist the selected unit. Command units are ignored, as it isn't trivial to take into account enhancements.

https://user-images.githubusercontent.com/15778155/193425009-2badf18b-4dc2-4077-977f-3d0f7a30a568.mp4

### CODE REVIEW

File: lua/keymap/misckeyactions.lua [hunk 1]
```diff
@@ -392,4 +393,77 @@ end
...
+    if selection then
+
+        -- try and find extractors that are upgrading
+        local upgrading = { }
+        for k, unit in selection do
+            if unit:GetWorkProgress() > 0 then
+                table.insert(upgrading, unit)
+            end
+
+            if next(upgrading) then
```
Comment (4z0t) [inline lua/keymap/misckeyactions.lua:416]: Why is this here?

Comment (Garanas) [reply lua/keymap/misckeyactions.lua:416]: to check if the array has at least one entry in it

Comment (4z0t) [reply lua/keymap/misckeyactions.lua:416]: Use table.empty instead. And why would you check it each iteration?

Comment (Hdt80bro) [reply lua/keymap/misckeyactions.lua:416]: `next` is more efficient, although I understand the concern about loss of conveyed intention. Although, if we cared about optimizing code, there are several other thing we could do here as well. As for why it is checked each iteration, it can change each iteration.

Comment (4z0t) [reply lua/keymap/misckeyactions.lua:416]: Selecting units is very expensive operation, I'd recommend do it in the end...

Comment (Garanas) [reply lua/keymap/misckeyactions.lua:416]: it is supposed to happen in the end.

**Comment by Garanas** [comment]
We'll tackle the 'get into transports' hotkey with another pull request
